packages/core/src/__integrationtests__/PublicCredentials.spec.ts

```diff
@@ -0,0 +1,315 @@
+  api = await initializeApi()
+}, 30_000)
+
+beforeAll(async () => {
+  tokenHolder = await createEndowedTestAccount()
+  attesterKey = makeSigningKeyTool()
+  attester = await createFullDidFromSeed(tokenHolder, attesterKey.keypair)
+}, 60_000)
+
+describe('When there is an attester and ctype NFT name', () => {
+  beforeAll(async () => {
+    const ctypeExists = await isCtypeOnChain(nftNameCType)
+    if (ctypeExists) return
+    const tx = await Did.authorizeTx(
+      attester.uri,
+      api.tx.ctype.add(CType.toChain(nftNameCType)),
+      attesterKey.getSignCallback(attester),
+      tokenHolder.address
+    )
+    await submitTx(tx, tokenHolder)
+  }, 60_000)
+
+  it('should be possible to issue a credential', async () => {
+    latestCredential = {
+      claims: {
+        name: `Certified NFT collection with id ${UUID.generate()}`,
+      },
+      cTypeHash: CType.getHashForSchema(nftNameCType),
+      delegationId: null,
+      subject: assetId,
+    }
+    const encodedPublicCredential = PublicCredential.toChain(latestCredential)
+    const storeTx = api.tx.publicCredentials.add(encodedPublicCredential)
+    const authorizedStoreTx = await Did.authorizeTx(
+      attester.uri,
+      storeTx,
+      attesterKey.getSignCallback(attester),
+      tokenHolder.address
+    )
+    await submitTx(authorizedStoreTx, tokenHolder)
+    const credentialId = PublicCredential.computeId(
+      latestCredential,
+      attester.uri
+    )
+
+    const publicCredentialEntry =
+      await api.call.publicCredentials.getCredential(credentialId)
+    expect(publicCredentialEntry.isSome).toBe(true)
+
+    const completeCredential = await PublicCredential.credentialFromChain(
+      credentialId,
+      publicCredentialEntry
+    )
+
+    // Verify that the retrieved credential matches the input one, plus the generated ID and the attester DID.
+    expect(completeCredential).toEqual(
+      expect.objectContaining({
+        ...latestCredential,
+        id: credentialId,
+        attester: attester.uri,
+        revoked: false,
+      })
+    )
+  })
+
+  it('should be possible to issue a second credential to the same asset and retrieve both of them', async () => {
+    latestCredential = {
+      claims: {
+        name: `Certified NFT collection with id ${UUID.generate()}`,
+      },
+      cTypeHash: CType.getHashForSchema(nftNameCType),
+      delegationId: null,
+      subject: assetId,
+    }
+    const encodedPublicCredential = PublicCredential.toChain(latestCredential)
+    const storeTx = api.tx.publicCredentials.add(encodedPublicCredential)
+    const authorizedStoreTx = await Did.authorizeTx(
+      attester.uri,
+      storeTx,
+      attesterKey.getSignCallback(attester),
+      tokenHolder.address
+    )
+    await submitTx(authorizedStoreTx, tokenHolder)
+
+    const encodedAssetCredentials =
+      await api.call.publicCredentials.getCredentials(assetId, null)
+    const assetCredentials = await PublicCredential.credentialsFromChain(
```

Comment:
Does having `encodedAssetCredentials` as a constant help? We’re only using it once, and it’s quite clear what the function returns.